openverse-api/catalog/api/utils/waveform.py

```diff
@@ -0,0 +1,131 @@
+import json
+import logging
+import math
+import mimetypes
+import os
+import pathlib
+import shutil
+import subprocess
+
+import requests
+
+log = logging.getLogger(__name__)
+
+TMP_DIR = pathlib.Path('/tmp').resolve()
+UA_STRING = 'OpenverseWaveform/0.0 (https://wordpress.org/openverse)'
+
+
+def ext_from_url(url):
+    """
+    Get the file extension from the given URL. Looks at the last part of the URL
+    path, and returns the string after the last dot.
+
+    :param url: the URL to the file whose extension is being determined
+    :returns: the file extension or ``None``
+    """
+    file_name = url.split('/')[-1]
+    if '.' in file_name:
+        ext = file_name.split('.')[-1]
+        return f'.{ext}'
+    else:
+        return None
+
+
+def download_audio(url, identifier):
+    """
+    Download the audio from the given URL to a location on the disk.
+
+    :param url: the URL to the file being downloaded
+    :param identifier: the identifier of the media object to name the file
+    :returns: the name of the file on the disk
+    """
+    log.info(f'Downloading file at {url}')
+
+    headers = {
+        'User-Agent': UA_STRING
+    }
+    with requests.get(url, stream=True, headers=headers) as res:
```

Comment:
What happens if the response comes back with a 404, if the audio file no longer exists for some reason? Is that even possible?

Reply:
At the top-most level, we have a `try`/`except` block with a catchall handler for any exception so there won't be a 500 scenario.